### PR TITLE
Allow Background Color and Background Image simultaneously on Canvas

### PR DESCRIPTION
**Pull request by thequixotic**
Currently Canvas does not support both Background Color and Image at the same time. This PR fixes that issue and allows both Background Color and Image simultaneously. 

Background Image is drawn on top of Background Color. 
The Companion side has been corrected.

### CODE REVIEW

File: appinventor/components/src/com/google/appinventor/components/runtime/Canvas.java [hunk 1]
```diff
@@ -91,7 +93,8 @@
     category = ComponentCategory.ANIMATION)
 @SimpleObject
 @UsesPermissions(permissionNames = "android.permission.INTERNET," +
-                 "android.permission.WRITE_EXTERNAL_STORAGE")
+                 "android.permission.WRITE_EXTERNAL_STORAGE," +
+                 "android.permission.CAMERA" )
```
Comment (jisqyv) [inline appinventor/components/src/com/google/appinventor/components/runtime/Canvas.java]: @thequixotic Why do you add CAMERA permission for this change?

(Sorry for the slow response, we are getting ready to do a Component release [first in a long time] and I want to get this change).

**Comment by thequixotic** [comment]
@jisqyv , I removed the Camera permission, it seems to have crept in from the Canvas Camera project.
Corrected now.

**Comment by afmckinney** [comment]
Great thanks.

On Wed, Jan 6, 2016, 9:35 AM thequixotic <notifications@github.com> wrote:

> @jisqyv <https://github.com/jisqyv> , I removed the Camera permission, it
> seems to have crept in from the Canvas Camera project.
> Corrected now.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/mit-cml/appinventor-sources/pull/584#issuecomment-169338310>
> .
>
-- 
--------------------------------------------------------------------------

Andrew F. McKinney • Director of Software Development • MIT App Inventor
Project • MIT, Center for Mobile Learning, Media Lab • MIT, Computer
Science and Artificial Intelligence Laboratory, CSAIL • mckinney@mit.edu •
978-255-3979 • http://appinventor.mit.edu • @MITAppInventor

**Comment by jisqyv** [comment]
LGTM (On Gerrit for the UCR branch)

**Comment by jisqyv** [comment]
@thequixotic MERGED (into ucr)

**Pull request closed without merging**